File: CharityWebBackend/src/main/java/group3/project/charityweb/service/DisbursementService.java
package group3.project.charityweb.service;

import group3.project.charityweb.model.dto.request.DisbursementRequest;
import group3.project.charityweb.model.dto.response.DisbursementResponse;
import org.springframework.data.domain.Page;

public interface DisbursementService {
    String createDisbursement(String username, String projectId, DisbursementRequest request);
    Page<DisbursementResponse> getProjectDisbursements(String projectId, int page, int size);
    DisbursementResponse getDisbursementById(String disbursementId);
}

File: CharityWebFrontend/src/services/DisbursementService.js
import api from './api';

export const createDisbursement = (projectId, data) =>
    api.post(`/projects/${projectId}/disbursements`, data);

export const getProjectDisbursements = (projectId) =>
    api.get(`/projects/${projectId}/disbursements`);

export const getDisbursementById = (disbursementId) =>
    api.get(`/disbursements/${disbursementId}`);
